presto-main/src/main/java/com/facebook/presto/cost/UnnestStatsRule.java

```diff
@@ -0,0 +1,82 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.cost;
+
+import com.facebook.presto.Session;
+import com.facebook.presto.cost.ComposableStatsCalculator.Rule;
+import com.facebook.presto.matching.Pattern;
+import com.facebook.presto.sql.planner.Symbol;
+import com.facebook.presto.sql.planner.TypeProvider;
+import com.facebook.presto.sql.planner.iterative.Lookup;
+import com.facebook.presto.sql.planner.plan.UnnestNode;
+
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+
+import static com.facebook.presto.sql.planner.plan.Patterns.unnest;
+
+public class UnnestStatsRule
```

Comment:
Please add a test (similar as we have for other rules, e.g.: JoinStatsRule, etc.). Also add a test to `TestDetermineJoinDistributionType` to make sure it works for good for your case.

Reply:
I added a `TestUnnestStatsRule`. Adding test to `TestDetermineJoinDistributionType` seems strange. It didn't work before because `DetermineJoinDistributionType` didn't get the stats from the side of the join with `Unnest`. The logic for `DetermineJoinDistributionType` doesn't really change. So I don't think adding test here is necessary nor correct.

Reply:
You can think of it as of an integration test. Just to verify that what you are trying to achieve work for good (for example for some weird reason the `DetermineJoinDistributionType` can still be getting unknown estimates for a table size for various reasons (e.g. missing nulls fraction, or whatever else)). If you feel strong that test like that is extra - i don't insist.

Reply:
You can think of it as of an integration test. Just to verify that what you are trying to achieve work for good (for example for some weird reason the `DetermineJoinDistributionType` can still be getting unknown estimates for a table size for various reasons (e.g. missing nulls fraction, or whatever else)). If you feel strong that test like that is extra - i don't insist.